image_client/picturae_importer.py

```diff
@@ -0,0 +1,919 @@
+
+        self.record_full = self.record_full.drop_duplicates(subset=['CatalogNumber'])
+
+        for index, row in self.record_full.iterrows():
+            self.populate_fields(row)
+            self.create_agent_list(row)
+            self.populate_taxon()
+            if self.taxon_id is None:
+                self.create_taxon()
+
+            if self.locality_id is None:
+                self.create_locality_record()
+
+            if len(self.new_collector_list) > 0:
+                self.create_agent_id()
+
+            self.create_collecting_event()
+
+            self.create_collection_object()
+
+            self.create_determination()
+
+            self.create_collector()
+
+
+    def upload_attachments(self):
+        """upload_attachments:
+                this function runs Botany importer
+                Updates date in
+                picturae_config to ensure prefix is
+                updated for correct filepath
+        """
+        try:
+            self.hide_unwanted_files()
+
+            BotanyImporter(paths=self.paths, config=picturae_config)
+
+            self.unhide_files()
+        except Exception as e:
+            self.logger.error(f"{e}")
+
+
+
+    def run_all_methods(self):
+        """run_all_methods:
+                        self-explanatory function, will run all methods in class in sequential manner"""
+        # code to create test images for test image uploads
+        # create_test_images(list(range(999999981, 999999985)), date_string=self.date_use)
+
+        # setting directory
+        to_current_directory()
+
+        self.assign_col_dtypes()
+
+        # creating file list after conditions
+        self.create_file_list()
+
+        # prompt
+        cont_prompter()
+
+        # locking users out from the database
+
+        sql = f"""UPDATE mysql.user
+                 SET account_locked = 'Y'
+                 WHERE user != '{picturae_config.USER}' AND host = '%';"""
+
+        insert_table_record(self.specify_db_connection, logger_int=self.logger, sql=sql)
+
+        # starting purge timer
+        self.exit_timestamp()
+
+        # creating tables
+
+        self.upload_records()
+
+        # creating new taxon list (should think of way to shorten this)
+        if len(self.taxon_list) > 0:
+            insert_taxa_added_record(taxon_list=self.taxon_list, connection=self.specify_db_connection,
+                                     logger_int=self.logger, df=self.record_full)
+
+        # uploading attachments
+        self.upload_attachments()
+
+        # writing time stamps to txt file
+
+        self.logger.info("process finished")
+
+        # unlocking database
+
+        sql = f"""UPDATE mysql.user
+                 SET account_locked = 'n'
+                 WHERE user != '{picturae_config.USER}' AND host = '%';"""
+
+        insert_table_record(self.specify_db_connection, logger_int=self.logger, sql=sql)
```

Comment:
PicturaeImport class shortened by about 100 lines, have plans to push further reductions in future PR. I think even more of these functions could be moved to util files.